Add docs for mode changes (#311)
saucelabs/sauce-docs: docs/testrunner-toolkit/running-tests.md

The following steps outline how to run your tests on your local machine with the containerized solution. This is the preferred option if you wish to run tests on your local machine, or if you wish to accelerate test execution in CI.

> Please note that if you don't specify a `--mode` flag when running tests, `saucectl` defaults to `sauce` mode.

Reviewer: I think this is no longer true. If you do not specify a runtime mode flag, the test will default to 1) the default config; or 2) the suite-specific config. It will ONLY default to `sauce` is none of those values is set.

The following steps outline how to run your tests on your local machine with the containerized solution. This is the preferred option if you wish to run tests on your local machine, or if you wish to accelerate test execution in CI.

:::note `docker` Syntax Page
See the [`docker` syntax reference page](/testrunner-toolkit/configuration/common-syntax#docker) for further details.
:::

:::note `mode` Description
See the [`mode` syntax reference page](/testrunner-toolkit/configuration/common-syntax#defaults) for further details.
:::

> For example, if you want to run all the suites in docker, the `config.yml` should look like so:

Reviewer: ```suggestion
For example, if you want to run all the suites in docker, set the default `mode` in `config.yml` to `docker`, as in the following example:
```

```yaml
defaults:
  mode: docker
```

### Specify a Docker Image

> `saucectl` can provide a default image to run your tests. You can also specify a Docker image in the `config.yml`. For example, if you set a Docker image should look like so:

Reviewer: ```suggestion
`saucectl` can provide a default image to run your tests. You can also specify a Docker image in the `config.yml`, as shown in the following example:
```

The following steps outline how to run your tests on your local machine with the containerized solution. This is the preferred option if you wish to run tests on your local machine, or if you wish to accelerate test execution in CI.

> Please note that if you don't specify a `--test-env` flag when running tests, `saucectl` defaults to mode settings in saucectl config file.

Reviewer: ```suggestion
Please note that if you don't specify a `--test-env` flag when running tests, `saucectl` defaults to `mode` settings in saucectl config file.
```

> Please note that if you don't specify a `--test-env` flag when running tests, `saucectl` defaults to `mode` settings in saucectl config file.

Reviewer: So I think we need to rewrite this whole section to direct users to use the config file to specify the `mode` for the project as `docker` rather than using the CLI flag.

<p><small>supported frameworks: <Highlight color="#25c2a0">cypress</Highlight></small></p>

> If you wish to run your tests on Sauce Labs VMs, simply set mode in `config.yml`:

Reviewer: ```suggestion
If you wish to run your tests on Sauce Labs VMs, set the default `mode` in `config.yml` to `sauce`, as shown in the following example:
```

### Run Tests in Docker Mode and Send Results to Sauce Labs

> Set docker mode in config like following:

Reviewer: ```suggestion
Set the default ` mode` in `config.yml` to `docker`:
```
